packages/fxa-auth-server/fxa-oauth-server/lib/config.js

```diff
@@ -349,34 +362,62 @@ var options = {
 
 conf.validate(options);
 
-// Replace openid key if file specified
+// Load our various keys from files if specified,
+// and check their contents.
 if (conf.get('openid.keyFile')) {
   conf.set('openid.key', require(conf.get('openid.keyFile')));
 }
 
+if (conf.get('openid.newKeyFile')) {
+  conf.set('openid.newKey', require(conf.get('openid.newKeyFile')));
+}
+
 if (conf.get('openid.oldKeyFile')) {
   conf.set('openid.oldKey', require(conf.get('openid.oldKeyFile')));
 }
 
-var key = conf.get('openid.key');
+// The active signing key should be a proper RSA private key.
+const key = conf.get('openid.key');
 assert.equal(key.kty, 'RSA', 'openid.key.kty must be RSA');
 assert(key.kid, 'openid.key.kid is required');
 assert(key.n, 'openid.key.n is required');
 assert(key.e, 'openid.key.e is required');
 assert(key.d, 'openid.key.d is required');
 
-var oldKey = conf.get('openid.oldKey');
-if (Object.keys(oldKey).length) {
+// The pending signing key, if present, should be a proper RSA private key
+// and must be different from the active key..
+const newKey = conf.get('openid.newKey');
+if (newKey && Object.keys(newKey).length) {
+  assert.equal(newKey.kty, 'RSA', 'openid.newKey.kty must be RSA');
+  assert(newKey.kid, 'openid.newKey.kid is required');
+  assert.notEqual(
+    key.kid,
+    newKey.kid,
+    'openid.key.kid must differ from openid.newKey.id'
+  );
+  assert(newKey.n, 'openid.newKey.n is required');
+  assert(newKey.e, 'openid.newKey.e is required');
+  assert(newKey.d, 'openid.newkey.d is required');
```

Comment:
We can't run the keys through Joi schemas?

Reply:
Good idea, added.